mirrord-layer/src/error.rs

```diff
@@ -53,3 +53,31 @@ pub enum LayerError {
     #[error("mirrord-layer: Failed inserting listen, already exists!")]
     ListenAlreadyExists,
 }
+
+//Todo: https://stackoverflow.com/questions/39150216/implementing-a-trait-for-multiple-types-at-once
+
+// what should the mapping be between LayerError and integer types
+
+// mapping based on - https://man7.org/linux/man-pages/man3/errno.3.html
+
+impl From<LayerError> for i32 {
+    fn from(error: LayerError) -> Self {
+        match error {
+            LayerError::VarError(_) => libc::EINVAL,
+            LayerError::ParseBoolError(_) => libc::EINVAL,
+            LayerError::SendErrorHookMessage(_) => libc::EBADMSG,
+            LayerError::SendErrorConnection(_) => libc::EBADMSG,
+            LayerError::SendErrorLayerTcp(_) => libc::EBADMSG,
+            LayerError::RecvError(_) => libc::EBADMSG,
+            LayerError::Null(_) => libc::EINVAL,
+            LayerError::TryFromInt(_) => libc::EINVAL,
+            LayerError::LocalFDNotFound(_) => libc::EBADF,
+            LayerError::EmptyHookSender => libc::ENOENT,
+            LayerError::NoConnectionId(_) => libc::ECONNREFUSED,
+            LayerError::IO(_) => libc::EIO,
```

Comment:
IO errors should probably use [`raw_os_error`](https://doc.rust-lang.org/std/io/struct.Error.html#method.raw_os_error). You could `unwrap_or` it to return `-1` for the `None` case.

Reply:
I dont think there is any point right now? where is this error being generated? it would always return None because as of now we are not using any `raw_os_error` related functions? if we do follow this, shouldn't we do the same for all errors